<a href="https://colab.research.google.com/github/Ukashashere/INTEL_AIOT_WORKSHOP/blob/main/Copy_of_01_hello_world_openvino.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
# Install openvino package
%pip install -q openvino

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 38.3/38.3 MB 29.9 MB/s eta 0:00:00



This is day_3 of AIOT_Intel workshop .The backend of Google Collab is a Linux machine. This is your boy Rockstar Ukasha.



In [ ]:
# Import openvino, check for its version and available plug-ins
import openvino as ov
print(ov.__version__)
core = ov.Core()
core


2023.3.0-13775-ceeafaf64f3-releases/2023/3


<Core: available plugins[CPU]>

In [ ]:
# Get the xml and bin file from Openvino toolkit storage

# Download Link : https://storage.openvinotoolkit.org/repositories/openvino_notebooks/models/mobelinet-v3-tf/FP32/


In [ ]:
!wget https://storage.openvinotoolkit.org/repositories/openvino_notebooks/models/mobelinet-v3-tf/FP32/v3-small_224_1.0_float.bin

--2024-03-06 06:26:49--  https://storage.openvinotoolkit.org/repositories/openvino_notebooks/models/mobelinet-v3-tf/FP32/v3-small_224_1.0_float.bin
Resolving storage.openvinotoolkit.org (storage.openvinotoolkit.org)... 3.163.125.118, 3.163.125.13, 3.163.125.65, ...
Connecting to storage.openvinotoolkit.org (storage.openvinotoolkit.org)|3.163.125.118|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 5075724 (4.8M) [application/octet-stream]
Saving to: ‘v3-small_224_1.0_float.bin’

v3-small_224_1.0_fl 100%[===================>]   4.84M  5.16MB/s    in 0.9s    

2024-03-06 06:26:51 (5.16 MB/s) - ‘v3-small_224_1.0_float.bin’ saved [5075724/5075724]



In [ ]:
!wget https://storage.openvinotoolkit.org/repositories/openvino_notebooks/models/mobelinet-v3-tf/FP32/v3-small_224_1.0_float.xml

--2024-03-06 06:26:54--  https://storage.openvinotoolkit.org/repositories/openvino_notebooks/models/mobelinet-v3-tf/FP32/v3-small_224_1.0_float.xml
Resolving storage.openvinotoolkit.org (storage.openvinotoolkit.org)... 3.163.125.118, 3.163.125.13, 3.163.125.65, ...
Connecting to storage.openvinotoolkit.org (storage.openvinotoolkit.org)|3.163.125.118|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 300608 (294K) [application/xml]
Saving to: ‘v3-small_224_1.0_float.xml’

v3-small_224_1.0_fl 100%[===================>] 293.56K   553KB/s    in 0.5s    

2024-03-06 06:26:55 (553 KB/s) - ‘v3-small_224_1.0_float.xml’ saved [300608/300608]



In [ ]:
# Load the model

model = core.read_model('/content/v3-small_224_1.0_float.xml')  #copied the path of xml file.
compiled_model = core.compile_model(model)
compiled_model

<CompiledModel:
inputs[
<ConstOutput: names[input:0] shape[1,224,224,3] type: f32>
]
outputs[
<ConstOutput: names[MobilenetV3/Predictions/Softmax:0] shape[1,1001] type: f32>
]>

#About the above output--

Output is determined for single image, we have "shape[1,..."
if we have 1000 image then "shape[1000,.." will be there



In [ ]:
# Compile the model

In [ ]:
!wget https://m.media-amazon.com/images/I/81Pp70CsSNL.jpg  -O demo.jpg  # copy image link here



--2024-03-06 06:54:46--  https://m.media-amazon.com/images/I/81Pp70CsSNL.jpg
Resolving m.media-amazon.com (m.media-amazon.com)... 18.164.156.97, 2600:9000:24ba:3c00:1d:d7f6:39d3:d9e1, 2600:9000:24ba:4c00:1d:d7f6:39d3:d9e1, ...
Connecting to m.media-amazon.com (m.media-amazon.com)|18.164.156.97|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 461502 (451K) [image/jpeg]
Saving to: ‘demo.jpg’

demo.jpg            100%[===================>] 450.69K  --.-KB/s    in 0.03s   

2024-03-06 06:54:46 (16.0 MB/s) - ‘demo.jpg’ saved [461502/461502]



In [ ]:
# Load the image and display
# DO preprocessing like resize and expand dimensions
import cv2
import numpy as np
img = cv2.imread('demo.jpg')
img = cv2.resize(img,(224,224))
img = np.expand_dims(img,0)  # we add 0 for the rows and here we expand dimensions
img.shape

# Now we've resized the image size according to what we've trained our model

(1, 224, 224, 3)

In [ ]:
# Infer the output

res_infer = compiled_model([img])[compiled_model.output(0)]
res_index = np.argmax(res_infer)
res_index

#

410

And the '410' value is corresponding to 'analog clock' in ---- https://gist.github.com/yrevar/942d3a0ac09ec9e5eb3a

In [ ]:
# Take the labels file and check the label name
# Download link : https://storage.openvinotoolkit.org/repositories/openvino_notebooks/data/data/datasets/imagenet/imagenet_2012.txt

In [ ]:
!wget https://storage.openvinotoolkit.org/repositories/openvino_notebooks/data/data/datasets/imagenet/imagenet_2012.txt

--2024-03-06 07:10:27--  https://storage.openvinotoolkit.org/repositories/openvino_notebooks/data/data/datasets/imagenet/imagenet_2012.txt
Resolving storage.openvinotoolkit.org (storage.openvinotoolkit.org)... 3.163.125.65, 3.163.125.84, 3.163.125.13, ...
Connecting to storage.openvinotoolkit.org (storage.openvinotoolkit.org)|3.163.125.65|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 31675 (31K) [text/plain]
Saving to: ‘imagenet_2012.txt’

imagenet_2012.txt   100%[===================>]  30.93K  --.-KB/s    in 0.1s    

2024-03-06 07:10:28 (235 KB/s) - ‘imagenet_2012.txt’ saved [31675/31675]



In [19]:
op = open('/content/imagenet_2012.txt').readlines()  #To change th number output further into redable output
op[res_index-1]            #index is subtracted to '-1' coz actual index starts at '0'

'n02708093 analog clock\n'